### DISCUSSION TITLE
DirSheet: Sort by name by default

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#2605] [Q&A] [answered]
This should be easy but I'm drawing a blank. I want all `DirSheet`s to open sorted by file name by default. How do I do this?

**Comment by midichef** [comment, marked as answer]
One quick way is to add to your .visidatarc:
`DirSheet._ordering = [('filename', False)]   # the boolean reflects the order of the sort; False = ascending, True = desscending`
But this relies on internal implementation details of the column sorting, not an outward-facing API, so it's not a clean long-term solution.

**Reply by cool-RR** [reply]
That is indeed quite disappointing. Works for me though, thank you.

**Comment by midichef** [comment]
Here is a better long-term solution, which does not depend on details of the current implementation. [Use `afterLoad` and `queueCommand`](https://github.com/saulpw/visidata/issues/1681#issuecomment-1783689728) in your .visidatarc:
```
@DirSheet.after
def afterLoad(sheet):
    vd.queueCommand('go-right', sheet=sheet)
    vd.queueCommand('sort-asc', sheet=sheet)
```
You should be able to use the same pattern for running startup commands on other sheets too, like `GraphSheet`, etc.

(EDIT:  use queueCommand instead of execCommand)

**Reply by cool-RR** [reply]
I've had [a bad experience](https://github.com/saulpw/visidata/issues/1681#issuecomment-1783873636) with `.after` so I'll pass. Thanks anyway.

**Reply by midichef** [reply]
Ah thanks, I hadn't seen that thread. In it saulpw stated that `queueCommand` is better than `execCommand` for general purposes, so I just edited my .visidatarc example snippet to reflect that.
